### PR TITLE
doc search

### PR DESCRIPTION
**Pull request by jjidt**
Only enabled for fetching draft docs from the projects/:id/documents route
Also, only currently searching text fields, but can be expanded with other models

### CODE REVIEW

**Review by dheniges — CHANGES_REQUESTED** [review]
Functionality is great, fuzzy search gives me fuzzy warm feelings. (Maybe the 2nd cup of coffee was a mistake)

I'd add a few more test cases:
- no match on any docs
- matching a text field value

**Comment by jjidt** [comment]
Now with more tests ™️

File: api/src/db/__testSetup__/factories/field.js [hunk 1]
```diff
@@ -12,7 +13,25 @@ const getFieldData = function(override = {}) {
   return { ...defaultProps, ...override }
 }
 
-const createMany = async function(numberOfFields, override) {
+const createMany = async function(numberOfFields, override = {} ) {
+  // default to a number field
+  const type = override.type || FIELD_TYPES.NUMBER
```
Comment (dheniges) [inline api/src/db/__testSetup__/factories/field.js]: tests are passing, but any concerns with changing this behavior to defaulting to number vs string field?

Comment (jjidt) [reply api/src/db/__testSetup__/factories/field.js]: we can do string, I just noticed that more tests were using numbers, I can change it back and make sure everything still works

**Review by dheniges — APPROVED** [review]
🌮

**Pull request merged**